scripts/timelockDeploy.ts

```diff
@@ -0,0 +1,94 @@
+import hre, { ethers } from 'hardhat';
+import NetworksForVerification from '@protocol/networksForVerification';
+import { TimelockController } from '@custom-types/contracts';
+import { expect } from 'chai';
+
+const oneDay = 24 * 60 * 60; /// 1 day timelock delay
+
+/// deploying the timelock on mainnet should look like this:
+///   MSIG=0xcBB83206698E8788F85EFbEeeCAd17e53366EBDf PROPOSERS=0xcBB83206698E8788F85EFbEeeCAd17e53366EBDf,0xA96D4a5c343d6fE141751399Fc230E9E8Ecb6fb6,0xB320e376Be6459421695F2b6B1E716AE4bc8129A,0xd90E9181B20D8D1B5034d9f5737804Da182039F6 EXECUTORS=0xcBB83206698E8788F85EFbEeeCAd17e53366EBDf \
+///   npx hardhat run --network mainnet scripts/timelockDeploy.ts
+
+/// deploying the timelock on arbitrum should look like this:
+///   MSIG=0x1A1075cef632624153176CCf19Ae0175953CF010 PROPOSERS=0x1A1075cef632624153176CCf19Ae0175953CF010,0xA96D4a5c343d6fE141751399Fc230E9E8Ecb6fb6,0xB320e376Be6459421695F2b6B1E716AE4bc8129A,0xd90E9181B20D8D1B5034d9f5737804Da182039F6 EXECUTORS=0x1A1075cef632624153176CCf19Ae0175953CF010 \
+///   npx hardhat run --network arbitrumOne scripts/timelockDeploy.ts
+
+/// Script does not hardcode msig, proposers or executors because this system will be deployed
+/// onto multiple networks where all of these addresses will change
+
+async function deploy(proposers: string[], executors: string[], multisig: string) {
+  const deployer = (await ethers.getSigners())[0];
+  const timelock = await (await ethers.getContractFactory('TimelockController')).deploy(oneDay, proposers, executors);
+  await timelock.deployed();
+
+  console.log(`\nTimelock deployed to: ${timelock.address}`);
+
+  const TIMELOCK_ADMIN_ROLE = await timelock.TIMELOCK_ADMIN_ROLE();
+  let tx = await timelock.grantRole(TIMELOCK_ADMIN_ROLE, multisig);
+  await tx.wait(1);
+
+  tx = await timelock.renounceRole(TIMELOCK_ADMIN_ROLE, deployer.address);
+  await tx.wait(1);
```

Comment:
Not positive on the syntax here, but here's another way to to waiting without making a temp variable.
```suggestion
  await (
    await timelock.renounceRole(TIMELOCK_ADMIN_ROLE, deployer.address)
  ).wait();
```